benchmarks/cpp/utils/generate.py

```diff
@@ -0,0 +1,316 @@
+                                                   random_source)
+        # Make sure it does not contain EOS token
+        eos_id = tokenizer.encode(tokenizer.eos_token, add_special_tokens=False)
+        while set(eos_id).issubset(sampled_ids):
+            tmp_id = (eos_id[0] + 1) % tokenizer.vocab_size
+            sampled_ids = [
+                tmp_id if element == eos_id[0] else element
+                for element in sampled_ids
+            ]
+        return sampled_ids
+
+    return [_sample(input_length) for input_length in input_lengths]
+
+
+def _generate_task_ids_uniform(
+    uniform_task_id_distribution: UniformTaskIdDistribution,
+    num_reqs: int,
+    random_source: random.Random,
+) -> Tuple[int, ...]:
+    return tuple(
+        random_source.randint(uniform_task_id_distribution.min_id,
+                              uniform_task_id_distribution.max_id)
+        for _ in range(num_reqs))
+
+
+def _generate_task_ids_constant(
+    constant_task_id_distribution: ConstantTaskIdDistribution,
+    num_reqs: int,
+) -> Tuple[int, ...]:
+    return tuple(constant_task_id_distribution.task_id for _ in range(num_reqs))
+
+
+def _generate_task_ids(
+    task_id_distribution: TaskIdDistribution,
+    num_reqs: int,
+    random_source: random.Random,
+) -> Tuple[int, ...]:
+    match task_id_distribution:
+        case UniformTaskIdDistribution() as uniform_task_id_distribution:
+            return _generate_task_ids_uniform(
+                uniform_task_id_distribution,
+                num_reqs,
+                random_source,
+            )
+        case ConstantTaskIdDistribution() as constant_task_id_distribution:
+            return _generate_task_ids_constant(
+                constant_task_id_distribution,
+                num_reqs,
+            )
+        case _:
+            raise ValueError(
+                f"Unsupported task id distribution: {task_id_distribution}")
+
+
+def generate_synthetic_text_dataset(
+    tokenizer: AutoTokenizer,
+    input_lengths_distribution: LengthDistribution,
+    output_lengths_distribution: LengthDistribution,
+    task_id_distribution: TaskIdDistribution,
+    num_reqs: int,
+    random_source: random.Random,
+) -> Workload:
+    input_lens = _generate_lengths(
+        input_lengths_distribution,
+        num_reqs,
+        random_source,
+    )
+    output_lens = _generate_lengths(
+        output_lengths_distribution,
+        num_reqs,
+        random_source,
+    )
+    task_ids = _generate_task_ids(
+        task_id_distribution,
+        num_reqs,
+        random_source,
+    )
+
+    max_input_len = max(input_lens)
+    max_output_len = max(output_lens)
+
+    input_ids = _generate_random_tokens(input_lens, tokenizer, random_source)
+
+    metadata = {
+        "num_requests": num_reqs,
+        "tokenize_vocabsize": tokenizer.vocab_size,
+        "max_input_len": max_input_len,
+        "max_output_len": max_output_len,
+    }
+
+    samples = [
+        TextSample(
```

Comment:
Something that this breaks is the ability for the dataset tool to write out text prompts. The `trtllm-bench` format allows either input ids or prompts. That allows us to reuse datasets when the scenario is the same.

Reply:
Ohhh I see what you mean. Okay that's easy to fix. We have 3 types of samples: ids, text and multi-modal instead of just 2. I'll fix that.

Reply:
Hmmm, after all maybe I don't see it. How was one supposed to generate plain text samples instead of tokenized ones before? I cannot find this logic in the old code somehow.

Reply:
Oh -- going back and looking I think you're right. I was thinking from when we get a dataset that has a prompt in it we might want to prefer the text version of that.